RecoCTPPS/TotemRPLocal/src/CTPPSDiamondRecHitProducerAlgorithm.cc

```diff
@@ -32,17 +31,19 @@ CTPPSDiamondRecHitProducerAlgorithm::build( const TotemRPGeometry* geom, const e
 
     edm::DetSet<CTPPSDiamondRecHit>& rec_hits = output.find_or_insert( detid );
 
-    for ( edm::DetSet<CTPPSDiamondDigi>::const_iterator digi = vec->begin(); digi != vec->end(); ++digi )
-    {
+    for ( edm::DetSet<CTPPSDiamondDigi>::const_iterator digi = vec->begin(); digi != vec->end(); ++digi ) {
+      if ( digi->getLeadingEdge()==0 and digi->getTrailingEdge()==0 ) { continue; }
+
       const int t = digi->getLeadingEdge();
-      if ( t==0 ) { continue; }
+      const int t0 = ( t-t_shift_ ) % 1024;
+      const int time_slice = ( t-t_shift_ ) / 1024;
 
-      const int t0 = ( t-t_shift_ ) % 1024,
-                time_slice = ( t-t_shift_ ) / 1024;
+      int tot = 0;
+      if ( t!=0 && digi->getTrailingEdge()!=0 ) tot = ( (int) digi->getTrailingEdge() ) - t;
```

Comment:
OK, I see the point that the hit can still be useful if either the leading or the trailing edge data is missing (=0)

Looking at a few digis:
```
en->Scan("EventAuxiliary.run():EventAuxiliary.luminosityBlock():EventAuxiliary.event():CTPPSDiamondDigiedmDetSetVector_ctppsDiamondRawToDigi_TimingDiamond_RECO.obj._sets.data.getLeadingEdge():CTPPSDiamondDigiedmDetSetVector_ctppsDiamondRawToDigi_TimingDiamond_RECO.obj._sets.data.getTrailingEdge()", "!(CTPPSDiamondDigiedmDetSetVector_ctppsDiamondRawToDigi_TimingDiamond_RECO.obj._sets.data.getLeadingEdge()==0&&CTPPSDiamondDigiedmDetSetVector_ctppsDiamondRawToDigi_TimingDiamond_RECO.obj._sets.data.getTrailingEdge()==0)")
***********************************************************************************
*    Row   * Instance * EventAuxi * EventAuxi * EventAuxi * CTPPSDiam * CTPPSDiam *
***********************************************************************************
*        0 *       25 *    296174 *        14 *  11207290 *      5093 *         0 *
*        1 *        1 *    296174 *        14 *  11205015 *      2401 *      1826 *
*        1 *        2 *    296174 *        14 *  11205015 *      4453 *      3893 *
*        1 *        3 *    296174 *        14 *  11205015 *      3437 *      2832 *
*        1 *        7 *    296174 *        14 *  11205015 *      2390 *      1844 *
*        1 *        9 *    296174 *        14 *  11205015 *         0 *      4888 *
*        1 *       10 *    296174 *        14 *  11205015 *         0 *      4864 *
*        1 *       25 *    296174 *        14 *  11205015 *      1993 *      2570 *
*        1 *       28 *    296174 *        14 *  11205015 *      3521 *         0 *
*        1 *       29 *    296174 *        14 *  11205015 *      3576 *      4100 *
*        1 *       32 *    296174 *        14 *  11205015 *      3017 *         0 *
*        1 *       33 *    296174 *        14 *  11205015 *      5036 *         0 *
```
compared to the rechits in the same events:
```
en->Scan("CTPPSDiamondRecHitedmDetSetVector_ctppsDiamondRecHits__RECO.obj._sets.data.getToT():CTPPSDiamondRecHitedmDetSetVector_ctppsDiamondRecHits__RECO.obj._sets.data.getT():CTPPSDiamondRecHitedmDetSetVector_ctppsDiamondRecHits__RECO.obj._sets.data.getOOTIndex()")
***********************************************************
*    Row   * Instance * CTPPSDiam * CTPPSDiam * CTPPSDiam *
***********************************************************
*        0 *        0 *         0 * 24.340820 *         4 *
*        1 *        0 * -14.03808 * 8.6181640 *         2 *
*        1 *        1 * -13.67187 * 8.7158203 *         4 *
*        1 *        2 * -14.77050 * 8.9111328 *         3 *
*        1 *        3 * -13.33007 * 8.3496093 *         2 *
*        1 *        4 *         0 *         0 *         0 *
*        1 *        5 *         0 *         0 *         0 *
*        1 *        6 * 14.086914 * 23.657226 *         1 *
*        1 *        7 *         0 * 10.961914 *         3 *
*        1 *        8 * 12.792968 * 12.304687 *         3 *
*        1 *        9 *         0 * 23.657226 *         2 *
*        1 *       10 *         0 * 22.949218 *         4 *
```
-  it seems pretty frequent to have the leading edge after the trailing edge. Does it make sense to fill ToT with a negative value in this case? It's not used in reco now. So, this is probably OK.
- it's also frequent to see 0 leading edge and non-zero trailing edge: does it make sense to assign OOTindex to 0 in this case?

Are these cases with out of order leading and trailing or either of them zero from significant noise in the system or there are some issues with the timing readout in this specific run/case?